Bug: when trying to hotreload config nil pointer panic occures

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

-->


**What happened**:

When trying to hotreload config nil pointer panic occures

**How to reproduce it (as minimally and precisely as possible)**:

Run nacos-farconf sample, and try to update the config stored in nacos, pixiu will try to hotreload the config file. At this time, nil pointer panic occures.

**Anything else we need to know?**:
log:

```
2025-05-28T11:42:44.952+0800	INFO	config/config_load.go:60	[dubbopixiu go] load path:../../dubbo-go-pixiu-samples/dubbogo/simple/nacos_farconf//pixiu/conf.yaml
2025-05-28T11:42:44.952+0800	INFO	config/config_load.go:86	load config in YAML format from : /mnt/c/Users/Alan.lxt-redmi/Desktop/repoistories/dubbo-go-pixiu-samples/dubbogo/simple/nacos_farconf/pixiu/conf.yaml
2025-05-28T11:42:45.038+0800	INFO	config/config_load.go:137	http config:map[idle_timeout:4s read_timeout:5s write_timeout:6s] true
2025-05-28T11:42:45.038+0800	INFO	cmd/gateway.go:208	GOMAXPROCS set to 18
2025-05-28T11:42:45.038+0800	INFO	server/pixiu_start.go:127	[dubbopixiu go] start by config : &{StaticResources:{Listeners:[0xc000799d90] Clusters:[] Adapters:[] ShutdownConfig:0xc000797aa0 PprofConf:{Enable:false Address:{SocketAddress:{Address:0.0.0.0 Port:8881 ResolverName: Domains:[] CertsDir:} Name:}}} DynamicResources:<nil> Metric:{Enable:false PrometheusPort:0} Node:<nil> Trace:<nil> Wasm:<nil> Config:0xc0004fb5c0 Nacos:0xc00064f2c0 Log:<nil>}
2025-05-28 11:42:45	INFO	config/root_config.go:138	[Config Center] Config center doesn't start
2025/05/28 11:42:45 Connected to 127.0.0.1:2181
2025-05-28T11:42:45.042+0800	INFO	proxy/descriptor_source.go:51	[dubbo-go-pixiu] could not load protoset files: no protoset files provided
2025-05-28T11:42:45.042+0800	INFO	tracing/driver.go:76	[dubbo-go-pixiu] no trace configuration in conf.yaml
2025-05-28T11:42:45.042+0800	INFO	http/http_listener.go:157	[dubbo-go-server] httpListener start at : 0.0.0.0:8881
2025/05/28 11:42:45 Authenticated: id=72057594730840067, timeout=4000
2025/05/28 11:42:45 Re-submitting `0` credentials after reconnect
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1172d34]

goroutine 89 [running]:
github.com/apache/dubbo-go-pixiu/pkg/hotreload.(*LoggerReloader).CheckUpdate(0x3749f60, 0xc00040dc00?, 0x0?)
	/mnt/c/Users/Alan.lxt-redmi/Desktop/repoistories/review/dubbo-go-pixiu/pkg/hotreload/logger.go:42 +0x74
github.com/apache/dubbo-go-pixiu/pkg/hotreload.(*Coordinator).hotReload(0x36f3f20, 0xc00040dc00)
	/mnt/c/Users/Alan.lxt-redmi/Desktop/repoistories/review/dubbo-go-pixiu/pkg/hotreload/hotreload.go:85 +0x97
github.com/apache/dubbo-go-pixiu/pkg/hotreload.(*Coordinator).HotReload(0x36f3f20)
	/mnt/c/Users/Alan.lxt-redmi/Desktop/repoistories/review/dubbo-go-pixiu/pkg/hotreload/hotreload.go:75 +0x52
created by github.com/apache/dubbo-go-pixiu/pkg/hotreload.StartHotReload in goroutine 1
	/mnt/c/Users/Alan.lxt-redmi/Desktop/repoistories/review/dubbo-go-pixiu/pkg/hotreload/hotreload.go:62 +0x7f
exit status 2
```

--- a/pkg/hotreload/logger.go
+++ b/pkg/hotreload/logger.go
@@ -18,9 +18,12 @@
 package hotreload
 
 import (
+	"go.uber.org/zap"
+)
+
+import (
 	"github.com/apache/dubbo-go-pixiu/pkg/logger"
 	"github.com/apache/dubbo-go-pixiu/pkg/model"
-	"go.uber.org/zap"
 )
 
 // LoggerReloader implements the HotReloader interface for reloading logger configurations.

--- a/pkg/hotreload/logger.go
+++ b/pkg/hotreload/logger.go
@@ -20,6 +20,7 @@ package hotreload
 import (
 	"github.com/apache/dubbo-go-pixiu/pkg/logger"
 	"github.com/apache/dubbo-go-pixiu/pkg/model"
+	"go.uber.org/zap"
 )
 
 // LoggerReloader implements the HotReloader interface for reloading logger configurations.
@@ -30,12 +31,12 @@ func (r *LoggerReloader) CheckUpdate(oldConfig, newConfig *model.Bootstrap) bool
 	oc := oldConfig.Log
 	nc := newConfig.Log
 
-	if oc == nil && nc != nil {
+	if nc == nil && oc != nil {
 		return true
-	}
-
-	if oc != nil && nc == nil {
+	} else if nc == nil && oc == nil {
 		return false
+	} else if nc != nil && oc == nil {
+		return true
 	}
 
 	// Check if any logger configuration fields have changed.
@@ -67,7 +68,11 @@ func (r *LoggerReloader) CheckUpdate(oldConfig, newConfig *model.Bootstrap) bool
 
 // HotReload applies the new logger configuration.
 func (r *LoggerReloader) HotReload(oldConfig, newConfig *model.Bootstrap) error {
-	if err := logger.HotReload(newConfig.Log.Build()); err != nil {
+	var newConfBuild *zap.Config
+	if newConfig.Log != nil {
+		newConfBuild = newConfig.Log.Build()
+	}
+	if err := logger.HotReload(newConfBuild); err != nil {
 		logger.Errorf("Failed to reload logger configuration: %v", err)
 		return err
 	}

